### DISCUSSION TITLE
Codespaces FAQ

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#122950] [Codespaces]
# Welcome to GitHub Codespaces in the Community Discussions!

[GitHub Codespaces](https://github.com/features/codespaces) provides cloud-powered development environments for any activity - whether it's a long-term project, or a short-term task like reviewing a pull request. You can work with these environments from Visual Studio Code or in a browser-based editor.

This discussion covers the typical questions and challenges related to GitHub Codespaces. We will keep it updated with latest FAQs and developments from the world of Codespaces. 

### 👋 Start Here 
-  [**GitHub Codespaces FAQs**](https://github.com/features/codespaces#:~:text=Frequently%20asked%C2%A0questions) ❔
-  [**Getting Started with GitHub Codespaces**](https://docs.github.com/en/codespaces) ⭐
-  [**Troubleshoot GitHub Codespaces**](https://docs.github.com/en/codespaces/troubleshooting) ⚙️ 

> [!Tip]
The resources above ⬆️  offer detailed information to address many common questions and troubleshooting issues regarding GitHub Codespaces. Prior to initiating a discussion, we encourage you to consult our guides. ❤️ 

-  If you're interested in exploring more discussions, providing product feedback, reporting bugs, or seeking answers, ▶️ [click here](https://github.com/orgs/community/discussions/categories/codespaces)
- For commonly asked questions for GitHub Codespaces in Visual Studio Code, see [Developing with GitHub Codespaces](https://code.visualstudio.com/docs/remote/codespaces#_common-questions)

---

# Codespaces FAQ 🔖 

## Where can I see my Codespaces for a single repository? ❓ 
You can see all existing Codespaces for a single repository by clicking the green "Code" button in the top right of your repository’s Code tab, above your repository files, and selecting the "Codespaces" tab. From here you can view and start any existing Codespaces in this repository, or create a new one.

## How do I access my Codespaces? ❓ 
You can see all your existing Codespaces across your account and organizations, or create a new one, on the [Codespaces dashboard](https://github.com/codespaces/).

## What clients can I use to access my Codespace? ❓
- [VS Code and VS Code Web](https://docs.github.com/en/codespaces/developing-in-codespaces/using-github-codespaces-in-visual-studio-code)
- [JetBrains IDEs](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-in-your-jetbrains-ide)
- [Work directly from the CLI](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-with-github-cli)

### Sync your settings
You can sync your editor settings between Codespaces, or Visual Studio Code on the desktop, by enabling [settings sync](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-github-codespaces-for-your-account#settings-sync). To enable settings sync click the cog icon at the bottom of the Activity Bar and select “Turn on Settings Sync.”

### Dotfiles
If you’ve already got a UNIX environment set up and want to share these settings across your Codespaces then you can use a [`dotfiles` repository](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-github-codespaces-for-your-account#dotfiles). Store these settings in a `dotfiles` repository and GitHub will clone this repository to your Codespace on creation.

## How do I help everyone using a project have a great experience in Codespaces? ❓
To configure dependencies, repository setup, and get services started automatically you’ll want to add a Development Container (devcontainer.json) to the repository. See the documentation for  [devcontainer.json in Codespaces](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers).

To decrease the time it takes to start up a codespace for a specific repository, see the documentation on  [prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-github-codespaces-prebuilds). 

> [!Note]
Codespaces is on by default for developers with a GitHub free account. If you belong to an organization, there may be a policy that prevents cloning—but if you can clone a repository, you will be able to start using Codespaces. Organizations will also need to pay for, enable, and manage their Codespaces instances.

## How much do Codespaces cost to use? ❓
Codespaces is free for individual use up to 60 hours a month and comes with simple, pay-as-you-go pricing after that. It’s also available for organizations with pay-as-you-go pricing and has pricing controls so any company or team can determine how much they want to spend a month. Learn more about Codespaces pricing for organizations [here](https://docs.github.com/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces) and the guide for [Troubleshooting Included Usage](https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-included-usage).

## How can I ensure I don’t spend too much? ❓
You can set a spending limit from your account settings. For more information see the [Managing the spending limit for GitHub Codespaces](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/managing-the-spending-limit-for-github-codespaces#managing-usage-and-spending-limit-email-notifications) documentation.

All organization administrators with access to billing reports can now see [projected codespaces spend](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#viewing-projected-usage-for-an-organization) in the month. 

## How can I see my current usage? ❓
You can check your current usage from [the billing page of your account settings](https://github.com/settings/billing). For more information check out the [Viewing your GitHub Codespaces usage](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/viewing-your-github-codespaces-usage) documentation.

## Is Codespaces available for students? ❓
Codespaces is available for free to students as part of the GitHub Student Developer Pack. Learn more about how to sign up and start using Codespaces and other GitHub products [here](https://education.github.com/pack). Also, Codespaces provides both maintainers and contributors with [generous free monthly usage](https://docs.github.com/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#monthly-included-storage-and-core-hours-for-personal-accounts). 

We recommend you checkout this discussion: [Making the most of your included free Codespaces usage](https://github.com/orgs/community/discussions/39697)

**Comment by Jackqhr** [comment]
I'm a user in China, and when i have been using it for long days. but these days when i try to connect the codespaces in southeast aisa, vut i fail to connect.when i try to connect the server in east us it works. so i guess there is something wrong with the server in southeast aisa.
![image](https://github.com/community/community/assets/43907194/44b39e6b-9b15-42be-96af-453f83fbaebe)

**Reply by wanitkapan** [reply]
> I'm a user in China, and when i have been using it for long days. but these days when i try to connect the codespaces in southeast aisa, vut i fail to connect.when i try to connect the server in east us it works. so i guess there is something wrong with the server in southeast aisa. ![image](https://private-user-images.githubusercontent.com/43907194/329551487-44b39e6b-9b15-42be-96af-453f83fbaebe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyMDIxODQsIm5iZiI6MTcxOTIwMTg4NCwicGF0aCI6Ii80MzkwNzE5NC8zMjk1NTE0ODctNDRiMzllNmItOWIxNS00MmJlLTk2YWYtNDUzZjgzZmJhZWJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI0VDA0MDQ0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1YWUzY2VkZDZhNTI5NzViODQ2Nzg0ZmQ1NmVjOTUwNGE4Y2RiOWI1ZjY4NWRlOWE5MGFjYzU1YjhkZDI5MGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.N4PI3fKH6R6bQ3qV44BFwJ5nGIEtO-I2JEc9ugxqTwQ)

**Reply by MorelliniMatteo** [reply]
It seems like you're experiencing regional connectivity issues when trying to connect to GitHub Codespaces in Southeast Asia, but not when connecting to the East US region. This might be due to several factors related to network routing, regional server stability, or restrictions that could impact access.

Possible Causes for Southeast Asia Server Connection Issues:
Regional Server Issues:

The Codespaces server in Southeast Asia may be experiencing temporary issues, outages, or high traffic, affecting its performance and connectivity. It’s possible that GitHub's infrastructure in this region is undergoing maintenance or facing disruptions.
Network Routing or Latency:

The network route between China and the Southeast Asia server could be facing instability or increased latency due to various reasons such as congested networks, restrictions, or bandwidth issues.
Internet Service Provider (ISP) Restrictions:

There could be issues related to your ISP that specifically affect connectivity to Southeast Asia servers. This could involve certain network blocks or routing problems that impact the path your connection takes.
Regional Firewalls or Restrictions:

In China, internet restrictions and the Great Firewall might impact connections to certain servers, depending on their region. However, it's interesting that your connection to the East US region works fine, which suggests that the issue is region-specific rather than global.
Troubleshooting Steps:
Try a Different Network:

If possible, test connecting to Codespaces in Southeast Asia from a different network (such as mobile data or a VPN). This could help identify whether the issue is specific to your current network or ISP.
Use a VPN:

You could try using a VPN with a Southeast Asia server to see if that improves connectivity. VPNs sometimes offer more stable routes, bypassing potential regional network restrictions.
Check GitHub Status Page:

Visit the GitHub Status Page to see if there are any ongoing outages or maintenance in the Southeast Asia region. This could give you more insight into whether GitHub is experiencing problems with its infrastructure there.
Contact GitHub Support:

You can also file a support ticket with GitHub to report the issue, specifying that you are having trouble connecting to Southeast Asia servers from China. They may provide insights or confirm if there are known issues affecting that region.
Use the East US Region Temporarily:

Since the East US server works fine for you, you can continue using it as a temporary solution. Although the latency might be slightly higher, it seems like a more stable connection for now.

**Reply by Belgarus** [reply]
It sounds like there might be connectivity issues between your location in China and GitHub Codespaces' Southeast Asia servers. Here are a few steps to troubleshoot and potentially resolve the issue:

1. Check Your Network and VPN
Since you are in China, network restrictions and firewalls might interfere with connections to certain regions.
If you're using a VPN, try switching to different locations and see if that helps.
If you're not using a VPN, try using one and connect to a different region.
2. Test Connectivity to GitHub Codespaces
Run the following command in your terminal to check network latency:
nginx
Kopieren
Bearbeiten
ping codespaces.githubusercontent.com
You can also run:
nginx
Kopieren
Bearbeiten
traceroute codespaces.githubusercontent.com
(or tracert on Windows) to see where the connection is getting blocked.
3. Switch to a Different Region
Since you mentioned that the East US server works, you might want to set your default region to East US in GitHub Codespaces settings.
Try launching a new Codespace and selecting a different region manually.
4. Check GitHub's Status Page
GitHub might be experiencing regional issues. You can check their status page for any reported outages: https://www.githubstatus.com/
5. Contact GitHub Support
If the issue persists, report it to GitHub support with details about your location, the affected region, and the error messages: https://support.github.com/

**Reply by andreymsdev** [reply]
t’s likely a routing problem or temporary instability in the Southeast Asia data centers. Since the East US server is working for you, it confirms your local connection is fine. You might want to stick with East US for now or check the [GitHub Status page](https://www.githubstatus.com/) to see if there's a reported outage in that region."

**Comment by pihedron** [comment]
The codespaces v2 connection always causes network errors for me. Why isn't there an option to revert back to v1?

I used to be able to disable v2 in the feature preview but now I don't have that choice. Now I have to migrate to Gitpod.

**Reply by MorelliniMatteo** [reply]
It sounds frustrating that you're encountering persistent network issues with GitHub Codespaces v2, especially when you had more stability with v1. While GitHub has introduced Codespaces v2 to improve performance and capabilities, it can lead to unexpected problems, like network errors, that are specific to individual environments or configurations.

Unfortunately, GitHub does not provide a rollback option to Codespaces v1, and they’ve moved forward with v2 as the primary version. The ability to toggle between versions (like in the feature preview) was likely part of a phased rollout to gather feedback before making v2 the default. As GitHub deprecates older versions, users are typically encouraged to transition to the newer one, which leaves fewer options to stay on the previous version.

Here are some potential reasons for your network errors and a few steps you could try:

1. Network Configuration Issues:
VPNs, proxies, or firewalls can sometimes cause connectivity problems with Codespaces. Ensure that your network allows the necessary traffic for Codespaces v2 to work properly.
Try running the Codespace in a different network (home vs office, or mobile data) to see if the problem persists.
2. Region-Specific Latency:
Codespaces uses cloud resources, and some regions may experience higher latency or connectivity issues. You could try creating your Codespace in a different region to check if the issue is region-specific.
3. Browser Compatibility:
If you’re using the browser-based editor, try switching browsers or ensuring your browser is up-to-date. Certain browser extensions may also interfere with Codespaces' connectivity, so try disabling them to see if that resolves the issue.
4. Open an Issue or Support Ticket:
Report your specific network errors to GitHub Support or open an issue on the GitHub Codespaces feedback forum. They may have visibility into widespread issues or configuration guidance based on your environment.
5. Gitpod as an Alternative:
If the issues with Codespaces persist and affect your workflow, transitioning to Gitpod might be a viable short-term alternative. Gitpod offers a similar cloud-powered development experience and can work across multiple platforms like GitHub and GitLab.

**Comment by wanitkapan** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> # Welcome to GitHub Codespaces in the Community Discussions!
> [GitHub Codespaces](https://github.com/features/codespaces) provides cloud-powered development environments for any activity - whether it's a long-term project, or a short-term task like reviewing a pull request. You can work with these environments from Visual Studio Code or in a browser-based editor.
> 
> This discussion covers the typical questions and challenges related to GitHub Codespaces. We will keep it updated with latest FAQs and developments from the world of Codespaces.
> 
> ### 👋 Start Here
> * [**GitHub Codespaces FAQs**](https://github.com/features/codespaces#:~:text=Frequently%20asked%C2%A0questions) ❔
> * [**Getting Started with GitHub Codespaces**](https://docs.github.com/en/codespaces) ⭐
> * [**Troubleshoot GitHub Codespaces**](https://docs.github.com/en/codespaces/troubleshooting) ⚙️
> 
> Tip
> 
> The resources above ⬆️ offer detailed information to address many common questions and troubleshooting issues regarding GitHub Codespaces. Prior to initiating a discussion, we encourage you to consult our guides. ❤️
> 
> * If you're interested in exploring more discussions, providing product feedback, reporting bugs, or seeking answers, ▶️ [click here](https://github.com/orgs/community/discussions/categories/codespaces)
> * For commonly asked questions for GitHub Codespaces in Visual Studio Code, see [Developing with GitHub Codespaces](https://code.visualstudio.com/docs/remote/codespaces#_common-questions)
> 
> # Codespaces FAQ 🔖
> ## Where can I see my Codespaces for a single repository? ❓
> You can see all existing Codespaces for a single repository by clicking the green "Code" button in the top right of your repository’s Code tab, above your repository files, and selecting the "Codespaces" tab. From here you can view and start any existing Codespaces in this repository, or create a new one.
> 
> ## How do I access my Codespaces? ❓
> You can see all your existing Codespaces across your account and organizations, or create a new one, on the [Codespaces dashboard](https://github.com/codespaces/).
> 
> ## What clients can I use to access my Codespace? ❓
> * [VS Code and VS Code Web](https://docs.github.com/en/codespaces/developing-in-codespaces/using-github-codespaces-in-visual-studio-code)
> * [JetBrains IDEs](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-in-your-jetbrains-ide)
> * [Work directly from the CLI](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-with-github-cli)
> 
> ## How do I make Codespaces feel like home? ❓
> ### Sync your settings
> You can sync your editor settings between Codespaces, or Visual Studio Code on the desktop, by enabling [settings sync](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-github-codespaces-for-your-account#settings-sync). To enable settings sync click the cog icon at the bottom of the Activity Bar and select “Turn on Settings Sync.”
> 
> ### Dotfiles
> If you’ve already got a UNIX environment set up and want to share these settings across your Codespaces then you can use a [`dotfiles` repository](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-github-codespaces-for-your-account#dotfiles). Store these settings in a `dotfiles` repository and GitHub will clone this repository to your Codespace on creation.
> 
> ## How do I help everyone using a project have a great experience in Codespaces? ❓
> To configure dependencies, repository setup, and get services started automatically you’ll want to add a Development Container (devcontainer.json) to the repository. See the documentation for [devcontainer.json in Codespaces](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers).
> 
> To decrease the time it takes to start up a codespace for a specific repository, see the documentation on [prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-github-codespaces-prebuilds).
> 
> Note
> 
> Codespaces is on by default for developers with a GitHub free account. If you belong to an organization, there may be a policy that prevents cloning—but if you can clone a repository, you will be able to start using Codespaces. Organizations will also need to pay for, enable, and manage their Codespaces instances.
> 
> ## How much do Codespaces cost to use? ❓
> Codespaces is free for individual use up to 60 hours a month and comes with simple, pay-as-you-go pricing after that. It’s also available for organizations with pay-as-you-go pricing and has pricing controls so any company or team can determine how much they want to spend a month. Learn more about Codespaces pricing for organizations [here](https://docs.github.com/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces) and the guide for [Troubleshooting Included Usage](https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-included-usage).
> 
> ## How can I ensure I don’t spend too much? ❓
> You can set a spending limit from your account settings. For more information see the [Managing the spending limit for GitHub Codespaces](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/managing-the-spending-limit-for-github-codespaces#managing-usage-and-spending-limit-email-notifications) documentation.
> 
> All organization administrators with access to billing reports can now see [projected codespaces spend](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#viewing-projected-usage-for-an-organization) in the month.
> 
> ## How can I see my current usage? ❓
> You can check your current usage from [the billing page of your account settings](https://github.com/settings/billing). For more information check out the [Viewing your GitHub Codespaces usage](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/viewing-your-github-codespaces-usage) documentation.
> 
> ## Is Codespaces available for students? ❓
> Codespaces is available for free to students as part of the GitHub Student Developer Pack. Learn more about how to sign up and start using Codespaces and other GitHub products [here](https://education.github.com/pack). Also, Codespaces provides both maintainers and contributors with [generous free monthly usage](https://docs.github.com/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#monthly-included-storage-and-core-hours-for-personal-accounts).
> 
> We recommend you checkout this discussion: [Making the most of your included free Codespaces usage](https://github.com/orgs/community/discussions/39697)

**Comment by wanitkapan** [comment]
Game 
if
Money
possible

**Comment by webmaster1225** [comment]
I used v1 for a long time before and any problems haven't caused. But some errors are caused in v2.

**Comment by Syammed2429** [comment]
Hi Guys, if anyone are facing any issue with the Codespace like it's auto terminating / exiting while running , you may can refer this thread
https://github.com/orgs/community/discussions/135598

**Comment by ehiber** [comment]
Hi I'm trying to open a new codespaces but I have this  issue and I don't know how to solve
![image](https://github.com/user-attachments/assets/21f3d090-b3e2-4d68-ba26-c5e6cebfa993)

**Reply by r-ish-i** [reply]
Hey! Did your issue got resolved?
If yes please mention the solution.

**Reply by nihalsheikh** [reply]
I am also encountering the same issue, did anyone find the solution?

**Comment by JorenVsyntra** [comment]
Hey everyone!

**Comment by TarnaWijaya** [comment]
This is great but why are extensions limited?

**Comment by JanGo-coding** [comment]
> # Welcome to GitHub Codespaces in the Community Discussions!
> [GitHub Codespaces](https://github.com/features/codespaces) provides cloud-powered development environments for any activity - whether it's a long-term project, or a short-term task like reviewing a pull request. You can work with these environments from Visual Studio Code or in a browser-based editor.
> 
> This discussion covers the typical questions and challenges related to GitHub Codespaces. We will keep it updated with latest FAQs and developments from the world of Codespaces.
> 
> ### 👋 Start Here
> * [**GitHub Codespaces FAQs**](https://github.com/features/codespaces#:~:text=Frequently%20asked%C2%A0questions) ❔
> * [**Getting Started with GitHub Codespaces**](https://docs.github.com/en/codespaces) ⭐
> * [**Troubleshoot GitHub Codespaces**](https://docs.github.com/en/codespaces/troubleshooting) ⚙️
> 
> Tip
> 
> The resources above ⬆️ offer detailed information to address many common questions and troubleshooting issues regarding GitHub Codespaces. Prior to initiating a discussion, we encourage you to consult our guides. ❤️
> 
> * If you're interested in exploring more discussions, providing product feedback, reporting bugs, or seeking answers, ▶️ [click here](https://github.com/orgs/community/discussions/categories/codespaces)
> * For commonly asked questions for GitHub Codespaces in Visual Studio Code, see [Developing with GitHub Codespaces](https://code.visualstudio.com/docs/remote/codespaces#_common-questions)
> 
> # Codespaces FAQ 🔖
> ## Where can I see my Codespaces for a single repository? ❓
> You can see all existing Codespaces for a single repository by clicking the green "Code" button in the top right of your repository’s Code tab, above your repository files, and selecting the "Codespaces" tab. From here you can view and start any existing Codespaces in this repository, or create a new one.
> 
> ## How do I access my Codespaces? ❓
> You can see all your existing Codespaces across your account and organizations, or create a new one, on the [Codespaces dashboard](https://github.com/codespaces/).
> 
> ## What clients can I use to access my Codespace? ❓
> * [VS Code and VS Code Web](https://docs.github.com/en/codespaces/developing-in-codespaces/using-github-codespaces-in-visual-studio-code)
> * [JetBrains IDEs](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-in-your-jetbrains-ide)
> * [Work directly from the CLI](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-with-github-cli)
> 
> ## How do I make Codespaces feel like home? ❓
> ### Sync your settings
> You can sync your editor settings between Codespaces, or Visual Studio Code on the desktop, by enabling [settings sync](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-github-codespaces-for-your-account#settings-sync). To enable settings sync click the cog icon at the bottom of the Activity Bar and select “Turn on Settings Sync.”
> 
> ### Dotfiles
> If you’ve already got a UNIX environment set up and want to share these settings across your Codespaces then you can use a [`dotfiles` repository](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-github-codespaces-for-your-account#dotfiles). Store these settings in a `dotfiles` repository and GitHub will clone this repository to your Codespace on creation.
> 
> ## How do I help everyone using a project have a great experience in Codespaces? ❓
> To configure dependencies, repository setup, and get services started automatically you’ll want to add a Development Container (devcontainer.json) to the repository. See the documentation for [devcontainer.json in Codespaces](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers).
> 
> To decrease the time it takes to start up a codespace for a specific repository, see the documentation on [prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-github-codespaces-prebuilds).
> 
> Note
> 
> Codespaces is on by default for developers with a GitHub free account. If you belong to an organization, there may be a policy that prevents cloning—but if you can clone a repository, you will be able to start using Codespaces. Organizations will also need to pay for, enable, and manage their Codespaces instances.
> 
> ## How much do Codespaces cost to use? ❓
> Codespaces is free for individual use up to 60 hours a month and comes with simple, pay-as-you-go pricing after that. It’s also available for organizations with pay-as-you-go pricing and has pricing controls so any company or team can determine how much they want to spend a month. Learn more about Codespaces pricing for organizations [here](https://docs.github.com/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces) and the guide for [Troubleshooting Included Usage](https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-included-usage).
> 
> ## How can I ensure I don’t spend too much? ❓
> You can set a spending limit from your account settings. For more information see the [Managing the spending limit for GitHub Codespaces](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/managing-the-spending-limit-for-github-codespaces#managing-usage-and-spending-limit-email-notifications) documentation.
> 
> All organization administrators with access to billing reports can now see [projected codespaces spend](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#viewing-projected-usage-for-an-organization) in the month.
> 
> ## How can I see my current usage? ❓
> You can check your current usage from [the billing page of your account settings](https://github.com/settings/billing). For more information check out the [Viewing your GitHub Codespaces usage](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/viewing-your-github-codespaces-usage) documentation.
> 
> ## Is Codespaces available for students? ❓
> Codespaces is available for free to students as part of the GitHub Student Developer Pack. Learn more about how to sign up and start using Codespaces and other GitHub products [here](https://education.github.com/pack). Also, Codespaces provides both maintainers and contributors with [generous free monthly usage](https://docs.github.com/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#monthly-included-storage-and-core-hours-for-personal-accounts).
> 
> We recommend you checkout this discussion: [Making the most of your included free Codespaces usage](https://github.com/orgs/community/discussions/39697)

**Comment by sharaddh** [comment]
I'm also encountering persistent network errors with the Codespaces v2 environment. Like @pihedron and @webmaster1225, I had a stable workflow with v1 and haven't faced these specific connectivity issues before.

It's frustrating that the option to disable v2 via feature preview is no longer available. Without a rollback option, it feels like we're forced to deal with instability that impacts daily development.

Has anyone found a reliable workaround for v2 specific networking issues, or is there a timeline for when these connection problems might be resolved? Considering alternatives like Gitpod is becoming a necessity for some of us until stability is restored.

**Comment by tanvishinde017** [comment]
Thanks for putting together this Codespaces FAQ. 🚀

A few additional tips that may help new users:

* **Stop inactive Codespaces** when you're finished working to avoid unnecessary core-hour usage.
* Use a **`devcontainer.json`** file to ensure everyone on the project gets the same dependencies and setup.
* Consider enabling **prebuilds** for repositories with lengthy setup processes—it can dramatically reduce startup times.
* Store secrets using **Codespaces Secrets** instead of committing credentials to the repository.
* If you're a student, don't forget that the **GitHub Student Developer Pack** includes generous Codespaces usage and many other benefits.
* For troubleshooting strange environment issues, rebuilding the container (`Codespaces: Rebuild Container`) often resolves dependency problems.

Personally, one of the biggest advantages of Codespaces is being able to switch between machines and continue working with the exact same development environment.

Thanks again for maintaining this discussion and keeping these resources up to date!

**Comment by SenShuvam** [comment]
I am from India using Codespace for the last 15 days. For the last couple of days, I have been regularly getting the message:
"There seems to be an issue authenticating with your GitHub token. Please visit https://cs50.dev/restart to restart your codespace and try again." The message is displayed if I use check50/submit50. I somehow got rectified yesterday, but now gaian it is there now. Kindly help.
